integration/rails_5.0_test.go

```diff
@@ -0,0 +1,155 @@
+
+		imageIDs[image.ID] = struct{}{}
+
+		container, err := settings.Docker.Container.Run.
+			WithEnv(map[string]string{
+				"PORT":            "8080",
+				"SECRET_KEY_BASE": "some-secret",
+			}).
+			WithPublish("8080").
+			WithPublishAll().
+			Execute(image.ID)
+		Expect(err).NotTo(HaveOccurred())
+
+		containerIDs[container.ID] = struct{}{}
+
+		Eventually(container).Should(BeAvailable())
+
+		response, err := http.Get(fmt.Sprintf("http://localhost:%s", container.HostPort("8080")))
+		Expect(err).NotTo(HaveOccurred())
+		defer response.Body.Close()
+
+		Expect(response.StatusCode).To(Equal(http.StatusOK))
+
+		content, err := ioutil.ReadAll(response.Body)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(string(content)).To(ContainSubstring("Hello World"))
+
+		Expect(logs).To(ContainLines(
+			MatchRegexp(fmt.Sprintf(`%s \d+\.\d+\.\d+`, settings.Buildpack.Name)),
+			"  Executing build process",
+			"    Running 'bundle exec rails assets:precompile assets:clean'",
+		))
+	})
+
+	context("when executing a rebuild", func() {
+		it("reuses the assets layer", func() {
+			build := settings.Pack.WithNoColor().Build.
+				WithBuildpacks(buildpacks...).
+				WithPullPolicy("never")
+
+			firstImage, firstLogs, err := build.Execute(name, source)
+			Expect(err).NotTo(HaveOccurred(), firstLogs.String())
+
+			imageIDs[firstImage.ID] = struct{}{}
+
+			Expect(firstImage.Buildpacks).To(HaveLen(6))
+			Expect(firstImage.Buildpacks[4].Key).To(Equal(settings.Buildpack.ID))
+			Expect(firstImage.Buildpacks[4].Layers).To(HaveKey("assets"))
+
+			container, err := settings.Docker.Container.Run.
+				WithCommand(fmt.Sprintf("ls -alR /layers/%s/assets/public/assets", strings.ReplaceAll(settings.Buildpack.ID, "/", "_"))).
+				Execute(firstImage.ID)
+			Expect(err).NotTo(HaveOccurred())
+
+			containerIDs[container.ID] = struct{}{}
+
+			secondImage, secondLogs, err := build.Execute(name, source)
+			Expect(err).NotTo(HaveOccurred(), secondLogs.String)
+
+			imageIDs[secondImage.ID] = struct{}{}
+
+			Expect(secondImage.Buildpacks).To(HaveLen(6))
+			Expect(secondImage.Buildpacks[4].Key).To(Equal(settings.Buildpack.ID))
+			Expect(secondImage.Buildpacks[4].Layers).To(HaveKey("assets"))
+
+			container, err = settings.Docker.Container.Run.
+				WithCommand(fmt.Sprintf("ls -alR /layers/%s/assets/public/assets", strings.ReplaceAll(settings.Buildpack.ID, "/", "_"))).
+				Execute(secondImage.ID)
+			Expect(err).NotTo(HaveOccurred())
+
+			containerIDs[container.ID] = struct{}{}
+
+			Expect(secondImage.Buildpacks[4].Layers["assets"].Metadata["built_at"]).To(Equal(firstImage.Buildpacks[4].Layers["assets"].Metadata["built_at"]))
+			Expect(secondImage.Buildpacks[4].Layers["assets"].Metadata["cache_sha"]).To(Equal(firstImage.Buildpacks[4].Layers["assets"].Metadata["cache_sha"]))
+
+			// TODO: why is the image id changing?
+			// Expect(secondImage.ID).To(Equal(firstImage.ID), fmt.Sprintf("%s\n\n%s", firstLogs, secondLogs))
```

Comment:
Do you still need this TODO?

Reply:
I want to keep this. I need to spend a little bit of time diffing the layers from both images to try to understand why the resulting IDs are different.